MM

monmoymaahdie opened the pull request:
created method calculateBMI that changes scene to a new screen where BMI calculation information can be inputted.

Commit:
Added button to calculate BMI and changes the scene. Started work on
input validaiton and formatting of the scenes to look nice

Commit:
updated bmi calc and fxml file

Commit:
Updated new scene to include ways for users to input their height and
weight for the BMI calculation

monmoymaahdie committed:
Merge branch 'master' into MM-InputValidation

jcharlpretorius commented on src/application/WorkoutSelectionView.fxml:
```diff
@@ -1,16 +1,16 @@
+<?import javafx.scene.text.Font?>
 
-<VBox fx:id="workoutSelection" alignment="CENTER" prefHeight="200.0" prefWidth="325.0" spacing="10.0" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1" fx:controller="application.WorkoutSelectionViewController">
+<VBox fx:id="workoutSelection" alignment="CENTER" prefHeight="200.0" prefWidth="325.0" spacing="10.0" xmlns="http://javafx.com/javafx/18" xmlns:fx="http://javafx.com/fxml/1" fx:controller="application.WorkoutSelectionViewController">
```
Did you make these changes for a reason or was it done automatically?
Notice the change from javafx/8 to javafx/18
The vm uses javafx/8, so I don't know if it will be compatible if we merge this

jcharlpretorius replied:
This 18 must be changed back to an 8.
This could cause problems with compatability. Gives me a warning the console:
 WARNING: Loading FXML document with JavaFX API of version 18 by JavaFX runtime of version 16

Commit:
Completed BMI scene with functional buttons. Calculate BMI will
calculates BMI (there is a placeholder for now), exit to main menu takes
user back to workout selection menu (first scene) and change unit button
should change scene to an identical scene except input units change
(there is a placeholder for now).

Commit:
Completed BMI calculator scene skeleton with working buttons.

Buttons do not carry out intended function (except exit button) yet but
are operational.
Calculate BMI -> calculates bmi (placeholder for now)
Change unit -> will change unit to imperial(placeholder for now)
Exit to main menu -> returns to original scene (This button functions as
intended)

Commit:
BMI calculator now works and can calculate BMI in both imperial and
metric units.

Inputs = Metric (kg,m)
		 Imperial (lbs,inches)

jcharlpretorius commented on src/application/WorkoutSelectionViewController.java:
```diff
@@ -199,11 +199,103 @@ void storeSets(Scene exerciseSelectionScene, ExerciseSets exercise, ArrayList<Te
     
+    double calculatedBMI = 0;
+    boolean isMetric = true;
+    Label calculatedBMILabel = new Label();
```
We shouldn't have non private/encapsulated instance variables in the controller class (or any class), and the calculate BMI method should really be separated into it's own class

Commit:
Changed version of javafx to 8 from 18

Was running javafx 18 on my machine. Changing version to avoid
compatability issues.
